files/blockinfile.py

```diff
@@ -0,0 +1,290 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2014, 2015 YAEGASHI Takeshi <yaegashi@debian.org>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+import re
+import os
+import tempfile
+
+DOCUMENTATION = """
+---
+module: blockinfile
+author:
+    - 'YAEGASHI Takeshi (@yaegashi)'
+extends_documentation_fragment:
+    - files
+    - validate
+short_description: Insert/update/remove a text block
+                   surrounded by marker lines.
+version_added: '2.0'
+description:
+  - This module will insert/update/remove a block of multi-line text
+    surrounded by customizable marker lines.
+options:
+  dest:
+    aliases: [ name, destfile ]
+    required: true
+    description:
+      - The file to modify.
+  state:
+    required: false
+    choices: [ present, absent ]
+    default: present
+    description:
+      - Whether the block should be there or not.
+  marker:
+    required: false
+    default: '# {mark} ANSIBLE MANAGED BLOCK'
+    description:
+      - The marker line template.
+        "{mark}" will be replaced with "BEGIN" or "END".
+  block:
+    aliases: [ content ]
+    required: false
+    default: ''
+    description:
+      - The text to insert inside the marker lines.
+        If it's missing or an empty string,
+        the block will be removed as if C(state) were specified to C(absent).
+  insertafter:
+    required: false
+    default: EOF
+    description:
+      - If specified, the block will be inserted after the last match of
+        specified regular expression. A special value is available; C(EOF) for
+        inserting the block at the end of the file.  If specified regular
+        expresion has no matches, C(EOF) will be used instead.
+    choices: [ 'EOF', '*regex*' ]
+  insertbefore:
+    required: false
+    default: null
```

Comment:
@jimi-c Thanks for the clarification, I'll fix it.  Please update [DOCUMENTATION.yml](https://github.com/ansible/ansible/blob/devel/examples/DOCUMENTATION.yml) to conform to that.  It would be nice if we would also have examples for `EXAMPLES` and `RETURN`.